src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -266,10 +276,16 @@ trait FastSync {
       requestedReceipts = requestedReceipts - handler
     }
 
+    private def validateBlocks(requestedHashes: Seq[ByteString], blockBodies: Seq[BlockBody]): Boolean = (requestedHashes zip blockBodies)
```

Comment:
this way we can reduce check time, If we encounter failing one we just stop checking (forall) if we want to filter out failing we have to check all of them
If we get failure it means that all block bodies may fail validation because they are from malicious peer or because of peer failure they are returned in incorrect order
those are the case I can think of, in my opinion failing fast in this case is better, but I can change that